core/lib/libtomcrypt/src/sub.mk

```diff
@@ -6,3 +6,7 @@ subdirs-$(_CFG_CORE_LTC_ACIPHER) += math
 subdirs-y += misc
 subdirs-y += modes
 subdirs-$(_CFG_CORE_LTC_ACIPHER) += pk
+
+ifneq ($(_CFG_CORE_LTC_ACIPHER),y)
+subdirs-$(_CFG_CORE_LTC_EC25519) += pk
+endif
```

Comment:
One single statement like this would be better IMO:
```make
ifneq (,$(filter y,$(_CFG_CORE_LTC_ACIPHER) $(_CFG_CORE_LTC_EC25519)))
subdirs-y += pk
endif
```

Slightly off-topic but perhaps we should consider allowing multiple occurrences of the same sub-directory? I mean the following change:
```diff
diff --git a/mk/subdir.mk b/mk/subdir.mk
index ffa61d4e..d280c309 100644
--- a/mk/subdir.mk
+++ b/mk/subdir.mk
@@ -21,6 +21,8 @@ srcs :=
 gen-srcs :=
 asm-defines-files :=
 
+uniq = $(if $1,$(firstword $1) $(call uniq,$(filter-out $(firstword $1),$1)))
+
 define process-subdir-srcs-y
 ifeq ($$(sub-dir),.)
 srcs                           += $1
@@ -174,8 +176,8 @@ this-out-dir :=
 asm-defines-y :=
 
 # Process subdirectories in current directory
-$$(foreach sd, $$(sub-subdirs), $$(eval $$(call process-subdir,$$(sd))))
+$$(foreach sd, $$(call uniq,$$(sub-subdirs)), $$(eval $$(call process-subdir,$$(sd))))
 endef #process-subdir
 
 # Top subdirectories
-$(foreach sd, $(subdirs), $(eval $(call process-subdir,$(sd))))
+$(foreach sd, $(call uniq,$(subdirs)), $(eval $(call process-subdir,$(sd))))
``` 
...would allow:
```make
subdirs-$(_CFG_CORE_LTC_ACIPHER) += pk
subdirs-$(_CFG_CORE_LTC_EC25519) += pk
```
Opinions welcome (I can create a separate PR)

Reply:
I like the idea @jforissier.